src/k8s/cmd/k8s/k8s_bootstrap.go

```diff
@@ -134,7 +136,13 @@ func newBootstrapCmd(env cmdutil.ExecutionEnvironment) *cobra.Command {
 				Config:  bootstrapConfig,
 				Timeout: opts.timeout,
 			})
-			if err != nil {
+			if errors.Is(err, context.DeadlineExceeded) {
+				// NOTE (hue): this might have failed because we're stuck (most likely indefinitely) in the pre-remove hook
+				// let's show the user a more helpful message
+				cmd.PrintErrf("Error: Failed to bootstrap the cluster.\n\nThe error was: %v\n\nIncreasing the timeout might help.\n", err)
+				env.Exit(1)
+				return
+			} else if err != nil {
```

Comment:
I see where you're going with this but 99% of the time if the deadline exceeded, the problem is *not* that the default timeout is too short. Usually something is broken.